tests/tracking/test_rest_tracking.py

```diff
@@ -1192,89 +1192,22 @@ def test_get_metric_history_bulk_interval_respects_max_results(mlflow_client):
     assert response_limited.json().get("metrics") == expected_metrics
 
 
-def test_get_metric_history_bulk_interval_calls_optimized_impl_when_expected(tmp_path):
```

Comment:
unfortunately there is no more optimized impl. previously, if min and max steps are provided, we didn't fetch any metric history at all, because we just determined the range from the min and max. 

however, this PR makes it such that we always fetch metric history (because it is necessary to correctly sample steps)